test/gtest/lppool.hpp

```diff
@@ -0,0 +1,338 @@
+        stride      = tensor<int64_t>{in_dim.size() - 2};
+        stride.data = lppool_config.stride;
+
+        norm_type = lppool_config.norm_type;
+
+        auto gen_input_value = [](auto...) {
+            return prng::gen_A_to_B<T>(static_cast<T>(-10.0f), static_cast<T>(10.0f));
+        };
+        input = tensor<T>{in_dim, in_strides}.generate(gen_input_value);
+
+        std::vector<size_t> out_dim;
+        if(in_dim.size() == 4)
+        {
+            OD      = std::floor(static_cast<float>(D - ksize[0]) / stride[0]) + 1;
+            OH      = std::floor(static_cast<float>(H - ksize[1]) / stride[1]) + 1;
+            out_dim = {N, C, OD, OH};
+        }
+        else
+        {
+            OD      = std::floor(static_cast<float>(D - ksize[0]) / stride[0]) + 1;
+            out_dim = {N, C, OD};
+        }
+
+        output = tensor<T>{out_dim};
+        std::fill(output.begin(), output.end(), std::numeric_limits<T>::quiet_NaN());
+
+        ref_output = tensor<T>{out_dim};
+        std::fill(ref_output.begin(), ref_output.end(), std::numeric_limits<T>::quiet_NaN());
+
+        input_dev  = handle.Write(input.data);
+        output_dev = handle.Write(output.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+        miopenStatus_t status;
+
+        auto dims = input.desc.GetNumDims();
+        if(dims == 4)
+        {
+            cpu_lppool_forward_2d<T>(
+                input, ref_output, N, C, D, H, OD, OH, ksize, stride, norm_type);
+        }
+        else if(dims == 3)
+        {
+            cpu_lppool_forward_1d<T>(input, ref_output, N, C, D, OD, ksize, stride, norm_type);
+        }
+        status = miopen::lppool::LPPoolForward(handle,
+                                               input.desc,
+                                               input_dev.get(),
+                                               output.desc,
+                                               output_dev.get(),
+                                               ksize[0],
+                                               dims == 4 ? ksize[1] : 1,
+                                               stride[0],
+                                               dims == 4 ? stride[1] : 1,
+                                               norm_type);
+        ASSERT_EQ(status, miopenStatusSuccess);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
+
+        auto error = miopen::rms_range(ref_output, output);
+
+        ASSERT_EQ(miopen::range_distance(ref_output), miopen::range_distance(output));
+        EXPECT_LT(error, threshold * 10);
+    }
+    LPPoolTestCase lppool_config;
+
+    tensor<T> input;
+    tensor<T> output;
+    tensor<T> ref_output;
+    tensor<int64_t> ksize;
+    tensor<int64_t> stride;
+
+    float norm_type;
+    int64_t N = 1, C = 1, D = 1, H = 1, OD = 1, OH = 1;
```

Comment:
should be `size_t` as in_dim is `vector<size_t>`

Reply:
they should be int64_t because kernel code use them as int64_t